AlphaWallet/WalletConnect/WalletConnectProvider.swift

```diff
@@ -258,6 +264,14 @@ extension WalletConnectProvider: WalletConnectServerDelegate {
             .eraseToAnyPublisher()
     }
 
+    private func verifyIfEip712v3And4(_ typedData: EIP712TypedData, walletSession: WalletSession) throws -> EIP712TypedData {
+        if let server = typedData.server, walletSession.server != server {
```

Comment:
Does this only work for WalletConnect v1? Or does it cover v2 too?

Context: https://github.com/AlphaWallet/alpha-wallet-ios/issues/6330#issue-1583535346

Reply:
it works for both